Engine/DataStream.cs

```diff
@@ -52,14 +52,13 @@ public DataStream(IDataFeed feed)
         {
             do
             {
-                TimeSlice timeSlice;
-                while (_feed.Bridge.TryDequeue(out timeSlice))
+                foreach (var timeSlice in _feed.Bridge.GetConsumingEnumerable())
                 {
                     AlgorithmTime = timeSlice.Time;
                     yield return timeSlice.Data;
                 }
-            }
-            while (!_feed.LoadingComplete && !_feed.Bridge.IsEmpty);
+            } 
+            while (!_feed.Bridge.IsCompleted);
```

Comment:
@mchandschuh Of course you're right :) 
GetConsumingEnumerable already has a while (!IsCompleted) {...} internally.

So the external loop would never actually loop. It is useless.